collections/src/main/scala/strawman/collection/convert/Wrappers.scala

```diff
@@ -426,7 +427,7 @@ private[collection] trait Wrappers {
 
     def iterator() = enumerationAsScalaIterator(underlying.keys) map (k => (k, underlying get k))
 
-    override def clear() = underlying.clear()
+    def clear() = iterator().foreach(entry => underlying.remove(entry._1))
```

Comment:
Interestingly this made me find this bug: we were previously calling `underlying.clear()`, whereas the `underlying` value (of type `java.util.Dictionary`) had no such method. It was implicitly converted to the same Scala wrapper class, infinitely calling `clear()` again…

This bug is also present in 2.12.